File: pinyougou-parent/pinyougou-shop-web/src/main/webapp/js/service/provincesService.js
//服务层
app.service('provincesService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../provinces/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../provinces/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findById=function(id){
		return $http.get('../provinces/findById.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../provinces/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../provinces/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../provinces/delete.php?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../provinces/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/ProvincesService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbProvinces;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ProvincesService {

	/**
	 * 返回全部列表
	 * @return
	 */
    List<TbProvinces> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
    PageResult findByPagination(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
    void add(TbProvinces provinces);
	
	
	/**
	 * 修改
	 */
    void update(TbProvinces provinces);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
    TbProvinces findById(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
    void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
    PageResult findWithConditionsByPagination(TbProvinces provinces, int pageNum, int pageSize);
	
}
